libs/sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -730,6 +730,28 @@ public Node visitResetGlobal(SqlBaseParser.ResetGlobalContext context) {
         return new ResetStatement<>(visitCollection(context.primaryExpression(), Expression.class));
     }
 
+    @Override
+    public Node visitSetSessionAuthorization(SqlBaseParser.SetSessionAuthorizationContext context) {
+        SetSessionAuthorizationStatement.Scope scope;
+        if (context.LOCAL() != null) {
+            scope = SetSessionAuthorizationStatement.Scope.LOCAL;
+        } else {
+            scope = SetSessionAuthorizationStatement.Scope.SESSION;
+        }
+
+        if (context.DEFAULT() != null) {
+            return new SetSessionAuthorizationStatement(scope);
+        } else {
+            var userName = ((StringLiteral) visit(context.username)).getValue();
```

Comment:
```suggestion
            var userLiteral = visit(context.username);
            assert userLiteral instanceof StringLiteral
                : "username must be a StringLiteral because the parser grammar is restricted to string literals";
            var userName = ((StringLiteral) userLiteral).getValue();
```

Just a suggestion to make the contract between AstBuilder and parser grammar more explicit.